### ISSUE TITLE
Ban infinite dividends

### ISSUE THREAD

**Issue by oscbyspro** [oscbyspro/Ultimathnum#69]
Banning infinite dividends seems like the simplest solution to (#68). It would also mean that all safe division inputs are well-behaved. Infinite dividends would be banned by setting the type to Finite\<T\>. Imagine the following with some conveniences:

```swift
protocol BinaryInteger {
    static func division(_ dividend: Finite<Self>, by divisor: Nonzero<Self>) -> Fallible<Division<Self, Self>>
}
```

**Comment by oscbyspro** [comment]
Infinite-by-infinite division is well-behaved but also uninteresting, so it's not a big loss. I could add nicher constraints than Finite\<T> and Natural\<T> but I would rather keep it simple whenever possible.

**Comment by oscbyspro** [comment]
I've prototyped a new division approach (#95). It seems a bit annoying to maintain, but otherwise viable. I could, perhaps, reduce it to just `Optional<T>` since the there's only one finite error case, `T.isSigned && a == T.min && b == -1`.

**Comment by oscbyspro** [comment]
I've had other ideas too. Like, I could strengthen the input guarantee such that only well-behaved divisions are allowed, basically hoisting all preconditions. That might be convoluted in another way, but I don't want to dismiss the idea just yet.

**Comment by oscbyspro** [comment]
One reason why I don't want to remove the `Fallible<T>` part is because of invariants like this:

```swift
I8(-128).quotient(Nonzero(-1)) == I8.exactly(IXL(-128) / IXL(-1))
```

**Comment by oscbyspro** [comment]
Back to the original post: divisions of infinite dividends are useless. A blanked ban might still be a viable alternative.

**Comment by oscbyspro** [comment]
Hm. I might start of by just fixing (#68) since I'm not fully convinced that the `Optional<Fallible<T>>` approach (#95) makes generic binary integer division simpler. One reason for this is that the failure modes are never simultaneously relevant. Like, an unsigned integer cannot produce an `error` and a signed integer cannot produce `nil`.

**Comment by oscbyspro** [comment]
I'll resolve this issue by tomorrow. The combinatorial maintenance hazard of implicit error propagation was the biggest blocker and it was resolves by using the new `sink` approach (#116) (#117).
